src/Deployment/PhpsecServer.php

```diff
@@ -52,8 +53,21 @@ public function connect(): void
 			$this->sftp->disconnect(); // @ may fail
 		}
 		$sftp = new SFTP($this->url['host'], $this->url['port'] ?? 22);
-		if (!$sftp->login(urldecode($this->url['user']), urldecode($this->url['pass']))) {
-			exit('Login Failed');
+		if (!$this->privateKey !== null) {
```

Comment:
Do you mean `if ($this->privateKey === null) {` ?

Reply:
should only be `if ($this->privateKey) {`